nf_core/pipelines/downloads/singularity.py

```diff
@@ -406,29 +402,29 @@ def fetch_containers(
                     # and copy from the cache to the output
                     containers_copy.append((container, cache_path, output_path))
                     total_tasks += 1
-                    self.progress.update(task, total=total_tasks)
+                    self.progress.update_main_task(total=total_tasks)
 
                 else:
                     # download or pull directly to the output
                     fetch_list.append((container, output_path))
 
         # Download all containers
         if containers_download:
-            self.progress.update(task, description="Downloading singularity images")
-            self.download_images(containers_download, task, parallel_downloads=4)
+            self.progress.update_main_task(description="Downloading singularity images")
+            self.download_images(containers_download, parallel_downloads=4)
```

Comment:
`parallel_downloads` is a CLI argument, currently stored in `self.parallel_downloads` of a `DownloadWorkflow` instance. I have not heard of anyone using it ever (except for debugging), but it is probably reasonable to keep it customizable? 

If not, we should remove the CLI argument.

Reply:
I didn't know there was a CLI argument ! I'll connect them back